packages/block-library/src/table/index.js

```diff
@@ -88,57 +103,51 @@ export const settings = {
 			{
 				type: 'raw',
 				selector: 'table',
-				schema: tableSchema,
+				schema: tablePasteSchema,
 			},
 		],
 	},
 
-	edit( { attributes, setAttributes, isSelected, className } ) {
-		const { content, hasFixedLayout } = attributes;
-		const toggleFixedLayout = () => {
-			setAttributes( { hasFixedLayout: ! hasFixedLayout } );
-		};
+	edit,
 
-		const classes = classnames(
-			className,
-			{
-				'has-fixed-layout': hasFixedLayout,
-			},
-		);
+	save( { attributes } ) {
+		const { hasFixedLayout, head, body, foot } = attributes;
+		const isEmpty = ! head.length && ! body.length && ! foot.length;
 
-		return (
-			<Fragment>
-				<InspectorControls>
-					<PanelBody title={ __( 'Table Settings' ) } className="blocks-table-settings">
-						<ToggleControl
-							label={ __( 'Fixed width table cells' ) }
-							checked={ !! hasFixedLayout }
-							onChange={ toggleFixedLayout }
-						/>
-					</PanelBody>
-				</InspectorControls>
-				<TableBlock
-					onChange={ ( nextContent ) => {
-						setAttributes( { content: nextContent } );
-					} }
-					content={ content }
-					className={ classes }
-					isSelected={ isSelected }
-				/>
-			</Fragment>
-		);
-	},
+		if ( isEmpty ) {
+			return null;
+		}
 
-	save( { attributes } ) {
-		const { content, hasFixedLayout } = attributes;
-		const classes = classnames(
-			{
-				'has-fixed-layout': hasFixedLayout,
-			},
-		);
+		const classes = classnames( {
+			'has-fixed-layout': hasFixedLayout,
+		} );
+
+		const Section = ( { type, rows } ) => {
+			if ( ! rows.length ) {
+				return null;
+			}
+
+			const Tag = `t${ type }`;
+
+			return (
+				<Tag>
+					{ rows.map( ( { cells }, rowIndex ) =>
+						<tr key={ rowIndex }>
+							{ cells.map( ( { content, tag }, cellIndex ) =>
+								<RichText.Content tagName={ tag } value={ content } key={ cellIndex } />
+							) }
+						</tr>
+					) }
+				</Tag>
+			);
+		};
 
 		return (
-			<RichText.Content tagName="table" className={ classes } value={ content } />
+			<table className={ classes }>
+				<Section type="head" rows={ head } />
+				<Section type="body" rows={ body } />
+				<Section type="foot" rows={ foot } />
+			</table>
 		);
```

Comment:
Will we need to add anything to the `deprecated` attribute or does this generate the same markup as before?

Reply:
In there that should still work. I'll test.

Reply:
Works for me! 👍

Reply:
This is cool given that is backward compatible 👍